### PR TITLE
Add links to troubleshooting docs

### PR DESCRIPTION
**Pull request by DaveCTurner**
Users facing problems with cluster coordination seemingly struggle to find the troubleshooting docs in the manual, without which it's hard to understand or resolve the problem leading to unnecessary support calls and escalations, typically at high urgency.

This commit introduces a central list of known (versioned) links to the relevant parts of the reference manual, and includes those links in the relevant error messages. There are certainly other places where this functionality would be valuable, but here we only focus on the troubleshooting guides for cluster coordination problems.

It does not currently validate that the links work correctly, but that would be a valuable followup. Experience has shown that it's better to have these links than not, even given the risk that they might be broken by a future change.

Closes #92741

### CODE REVIEW

**Review by andreidan — APPROVED** [review]
LGTM, thanks for adding this David

File: server/src/main/java/org/elasticsearch/common/ReferenceDocs.java [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+            ? UNRELEASED_VERSION_COMPONENT
+            : currentVersion.major + "." + currentVersion.minor;
+    }
+
+    @Override
+    public String toString() {
+        return "https://www.elastic.co/guide/en/elasticsearch/reference/" + VERSION_COMPONENT + "/" + relativePath;
+    }
+
+    public static List<ReferenceDocs> linksToVerify(Version currentVersion, boolean isSnapshot) {
```
[the remark this answers is not in the document]
Comment (andreidan) [reply server/src/main/java/org/elasticsearch/common/ReferenceDocs.java:45]: ++ on Javadoc. Thanks David

[the remark this answers is not in the document]
Comment (DaveCTurner) [reply server/src/main/java/org/elasticsearch/common/ReferenceDocs.java]: Done, see [7f7a048](https://github.com/elastic/elasticsearch/pull/92755/commits/7f7a04830a8b19c3d9ffe536af2939ea4b7c6b10)

**Review by not-napoleon — APPROVED** [review]
This looks in line with what I was thinking.  At some point, we'll need to think about what happens if/when content moves in the docs, but this leaves us a good amount of flexibility to address that when we get to it.

**Pull request merged**